stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/SupportedReactions.kt

```diff
@@ -0,0 +1,124 @@
+package io.getstream.chat.android.ui
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import androidx.core.content.ContextCompat
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactionTypes.LOL
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactionTypes.LOVE
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactionTypes.THUMBS_DOWN
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactionTypes.THUMBS_UP
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactionTypes.WUT
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactions.lolDrawable
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactions.loveDrawable
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactions.thumbsDownDrawable
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactions.thumbsUpDrawable
+import io.getstream.chat.android.ui.SupportedReactions.DefaultReactions.wutDrawable
+import io.getstream.chat.android.ui.common.extensions.internal.getColorCompat
+
+/**
+ * Class allowing to define set of supported reactions
+ */
+public class SupportedReactions(
+    context: Context,
+    /**
+     * Map with keys corresponding to reaction type, value corresponding to Drawable Int.
+     * By default it's initialized with standard reactions. You can customize reactions by providing your own
+     * @property reactions: Map<String, ReactionDrawable> instance.
+     */
+    public val reactions: Map<String, ReactionDrawable> = mapOf(
+        LOVE to loveDrawable(context),
+        THUMBS_UP to thumbsUpDrawable(context),
+        THUMBS_DOWN to thumbsDownDrawable(context),
+        LOL to lolDrawable(context),
+        WUT to wutDrawable(context),
+    ),
+) {
+    public val types: List<String> = reactions.keys.toList()
+
+    internal fun isReactionTypeSupported(type: String): Boolean {
+        return reactions.keys.contains(type)
+    }
+
+    public fun getReactionIconStateInactive(type: String): Drawable? {
+        return reactions[type]?.inactiveDrawable
+    }
+
+    public fun getReactionIconStateActive(type: String): Drawable? {
+        return reactions[type]?.activeDrawable
+    }
+
+    public fun getReactionDrawable(type: String): ReactionDrawable? {
+        return reactions[type]
+    }
+
+    public class ReactionDrawable(public val inactiveDrawable: Drawable?, public val activeDrawable: Drawable?)
+
+    /**
+     * Default reaction types
+     */
+    public object DefaultReactionTypes {
```

Comment:
IMO it might be worth it to move it inside `DefaultReactions` object and just call it `Type`. What do you think?